src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/LicenceRepository.kt

```diff
@@ -11,4 +12,26 @@ interface LicenceRepository : JpaRepository<Licence, Long>, JpaSpecificationExec
   fun findAllByNomsIdAndStatusCodeIn(nomsId: String, status: List<LicenceStatus>): List<Licence>
   fun findAllByCrnAndStatusCodeIn(crn: String, status: List<LicenceStatus>): List<Licence>
   fun findByStatusCodeAndProbationAreaCode(statusCode: LicenceStatus, probationAreaCode: String): List<Licence>
+
+  @Query(
+    """
+    SELECT new uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.UnapprovedLicence( l.crn, l.forename, l.surname, com.firstName , com.lastName , com.email)
+        FROM Licence l 
+        JOIN l.submittedBy com
+        WHERE (l.actualReleaseDate = CURRENT_DATE OR l.conditionalReleaseDate = CURRENT_DATE) 
+        AND l.statusCode = 'SUBMITTED'
+        AND EXISTS 
+                (SELECT 1 FROM AuditEvent ae WHERE l.id = ae.licenceId AND ae.detail LIKE '%APPROVED%')
```

Comment:
as discussed this is a correlated subquery that could have performance issues with large numbers of matching licences, but think it's a lot more readable than an alternative, it also sounds like its a bit of a stop gap. 

It's also possibly missing indexes

Reply:
This is far from idea but currently we have no other way of knowing if a pre active licence has been previously Approved. There is another ticket to implement versioning on pre Active licences. Once complete, this check will no longer be needed.

The only other method would be to join on the audit table with an inline AND statement. It would require grouping on the licence to avoid duplicates rows brought back by each audit row. 

I don't think performance would be much different.